### PR TITLE
Clean code related to signal caster

### PR DESCRIPTION
**Pull request by jmirabel**
This fixes https://github.com/stack-of-tasks/dynamic-graph/issues/61 and clean code.

### CODE REVIEW

**Comment by jmirabel** [comment]
I haven't tested this with the whole stack.

**Comment by olivier-stasse** [comment]
Thanks Joseph, this looks much simpler than the complex caster system.
It is a nice work.
At this stage if the unit tests are passing and that we do not decrease the coverage percentage it sounds quite good.
If you think we should be more cautious we can put it on a branch and perform integration tests, or do it directly on devel and if needed revert.
What do you prefer ?

File: tests/signal-cast-registerer.cpp [hunk 1]
```diff
@@ -140,13 +140,11 @@ BOOST_AUTO_TEST_CASE(custom_vector_registerer) {
   dynamicgraph::Signal<dynamicgraph::Vector, int> myVectorSignal("vector");
 
   /// Create a second local vector registerer to generate an exception.
-  bool res = false;
   try {
     EigenCastRegisterer_V myVectorCast2;
   } catch (const ExceptionSignal &aes) {
-    res = (aes.getCode() == ExceptionSignal::GENERIC);
+    // BOOST_CHECK(aes.getCode() == ExceptionSignal::GENERIC);
```
[the remark this answers is not in the document]
Comment (jmirabel) [reply tests/signal-cast-registerer.cpp]: I think you commented an outdated change. I think it has been removed.

**Comment by jmirabel** [comment]
> @jmirabel Can we merge it ?

Some dependent packages must be updated too and the updates are mixed with the switch to boost python (and thus c++ 11)...

**Comment by jmirabel** [comment]
I'll close this PR and merge its content into branch `cpp11`. We can open PRs from cpp11 to devel later, when ready.

**Pull request closed without merging**